File: web-logger/web-logger.py
class WebLogger:
    
    def __init__(self):
        # initialize your data structure here.
        self.q = []
    
    def pop_q(self, timestamp):
        timestamp -= 299
        while len(self.q) and self.q[0] < timestamp:
            self.q.pop(0)

# @param {int} timestamp an integer
# @return nothing
def hit(self, timestamp):
    # Write your code here
    self.q.append(timestamp)
        self.pop_q(timestamp)
    
    
    # @param {int} timestamp an integer
    # @return {int} an integer
    def get_hit_count_in_last_5_minutes(self, timestamp):
        # Write your code here
        self.pop_q(timestamp)
        return len(self.q)

File: web-logger/web-logger.cc
class WebLogger {
    public:
    WebLogger() {
        // initialize your data structure here.
    }
    
    void pop_hits(int ts) {
        ts -= 299;
        while (!hits.empty() && hits.front() < ts) {
            hits.pop();
        }
    }
    
    /**
     * @param timestamp an integer
     * @return void
     */
    void hit(int timestamp) {
        // Write your code here
        hits.push(timestamp);
        pop_hits(timestamp);
    }
    
    /**
     * @param timestamp an integer
     * @return an integer
     */
    int get_hit_count_in_last_5_minutes(int timestamp) {
        // Write your code here
        pop_hits(timestamp);
        return hits.size();
    }
    
    private:
    queue<int> hits;
};
